reproman/interface/docker.shim

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/perl
+#emacs: -*- mode: shell-script; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: t -*-
+#ex: set sts=4 ts=4 sw=4 noet:
+#
+#
+# This script is a shim for the docker executable. The shim intercepts
+# calls to the docker executable, runs a image id discovery process, and
+# then proceeds with the original call to the original docker executable.
+#
+require 5.002;
+
+# Print debugging if docker debug switch is set
+my $DEBUG = 0;
+foreach(@ARGV) {
+    $DEBUG = 1 if ($_ eq '--debug' or $_ eq '-D');
+}
+
+# Print debug output
+# 
+# Parameters
+# ----------
+# $output
+#     string : Text to print to the screen.
+# 
+sub debug {
+    return if (!$DEBUG);
+    foreach my $line (@_) {
+        print "[DEBUG DOCKER SHIM] ${line}";
+        print "\n" if (!($line =~ /\n$/));
+    }
+}
+
+# Run commands in shell
+# 
+# Parameters
+# ----------
+# $command
+#     string : Command to execute in the shell
+# $default_message
+#     string : Message to send if nothing is found in command stderr
+#
+# Returns
+# -------
+# string : stdout of the executed command
+# 
+sub exec_handler {
+
+	my $command = shift;
+	my $default_message = shift;
+
+	my $output = qx/$command 2>&1/;
+	if ($? != 0) {
+		if ($output) {
+			die $output
+		}
+		else {
+  		die $default_message;
```

Comment:
nitpick: Shouldn't that else's `die` be indented?  But the above is short/simple enough that the if/else block could be converted to a single line.